Columns/Provider.php

```diff
@@ -50,21 +48,39 @@ public function onNewVisit(Request $request, Visitor $visitor, $action)
             return false;
         }
 
-        $ip = $visitor->getVisitorColumn('location_ip');
+        $userInfo = $this->getUserInfo($request, $visitor);
```

Comment:
Does it makes sense to use this information if the same logic is in UserCountry/Columns/Provider.php? If users want to use this information, they can just disable the Provider plugin, and if they want the actual reverse IP lookup, they could enable Provider?

Reply:
The database column is currently defined in the Provider plugin. The one in core does not hold the definition, so it would not be created unless the plugin is in use. Also the core dimension only overwrites the value if the the Provider plugin is enabled. So that part in Core only worked together with the Provider plugin, so it makes sense to simply move it here...

Reply:
Would it make sense to move the column definition & main logic to core? Then the Provider plugin would just be for if the user wants to the reverse IP lookup. It doesn't make sense for the main column definition to be outside of core, yet the preferred logic for setting the column to be in a plugin that we do not enable by default. Unless we don't want the column to be added if the provider plugin is not enabled, in which case we can remove the UserCountry column?

Reply:
@diosmosis the column from core would be removed here: https://github.com/matomo-org/matomo/pull/16756
If we leave the column in core, that actually wouldn't help anyone. The reports and additions for visits log & profile are defined in the Provider plugin. So the data would only be tracked, but not shown anywhere

Reply:
I didn't see that other PR, I understand now.